orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt

```diff
@@ -427,6 +451,130 @@ class SqlExecutionRepository(
       )
     }
 
+  override fun retrievePipelineConfigIdsForApplication(application: String): List<String> =
+    withPool(poolName) {
+      return jooq.selectDistinct(field("config_id"))
+        .from(PIPELINE.tableName)
+        .where(field("application").eq(application))
+        .fetch(0, String::class.java)
+    }
+
+  /**
+   * this function supports the following ExecutionCriteria currently:
+   * 'limit', a.k.a page size and
+   * 'statuses'.
+   *
+   * It executes the following query to determine how many pipeline executions exist that satisfy the above
+   * ExecutionCriteria. It then returns a list of all these execution ids.
+   *
+   * It does this by executing the following query:
+   * - If the execution criteria does not contain any statuses:
+   *    SELECT config_id, id
+        FROM pipelines force index (`pipeline_application_idx`)
+        WHERE application = "myapp"
+        ORDER BY
+        config_id;
+   * - If the execution criteria contains statuses:
+   *    SELECT config_id, id
+        FROM pipelines force index (`pipeline_application_status_starttime_idx`)
+        WHERE (
+          application = "myapp" and
+          status in ("status1", "status2)
+        )
+        ORDER BY
+        config_id;
+
+   * It then applies the limit execution criteria on the result set obtained above. We observed load issues in the db
```

Comment:
Looking at this - and looking at the indexes, that's I SUSPECT because there's no index with config_id, id application and status.  I think a better index here would help a LOT on narrowing this down & performing better.

Reply:
Can we get performance benchmarks on this?  PARTICULARLY if we re-ran those benchmarks with different indexes created and NOT having the force index section?

Reply:
I don't think so.  This is one of the great struggles of taking so long to get stuff pushed upstream.  Any data we may have had has vanished.  Plus also, we've since moved on to use the read replica for this, so collecting new data is gonna be tough.